usd/src/Conversions.cc

```diff
@@ -0,0 +1,154 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "sdf/usd/Conversions.hh"
+
+#include <ignition/common/Pbr.hh>
+
+#include "sdf/Pbr.hh"
+
+namespace sdf
+{
+  inline namespace SDF_VERSION_NAMESPACE {
+  //
+  namespace usd
+  {
+  sdf::Material convert(const std::shared_ptr<ignition::common::Material> &_in)
+  {
+    sdf::Material out;
+    out.SetEmissive(_in->Emissive());
+    out.SetDiffuse(_in->Diffuse());
+    out.SetSpecular(_in->Specular());
+    out.SetAmbient(_in->Ambient());
+    out.SetRenderOrder(_in->RenderOrder());
+    out.SetLighting(_in->Lighting());
+    out.SetDoubleSided(_in->TwoSidedEnabled());
+    const ignition::common::Pbr * pbr = _in->PbrMaterial();
+    if (pbr != nullptr)
+    {
+      out.SetNormalMap(pbr->NormalMap());
+      sdf::Pbr pbrOut;
+      sdf::PbrWorkflow pbrWorkflow;
+      pbrWorkflow.SetAlbedoMap(pbr->AlbedoMap());
+      pbrWorkflow.SetMetalnessMap(pbr->MetalnessMap());
+      pbrWorkflow.SetEmissiveMap(pbr->EmissiveMap());
+      pbrWorkflow.SetRoughnessMap(pbr->RoughnessMap());
+      pbrWorkflow.SetSpecularMap(pbr->SpecularMap());
+      pbrWorkflow.SetEnvironmentMap(pbr->EnvironmentMap());
+      pbrWorkflow.SetAmbientOcclusionMap(pbr->AmbientOcclusionMap());
+      pbrWorkflow.SetLightMap(pbr->LightMap());
+      pbrWorkflow.SetRoughness(pbr->Roughness());
+      pbrWorkflow.SetGlossiness(pbr->Glossiness());
+      pbrWorkflow.SetMetalness(pbr->Metalness());
+
+      if (pbr->NormalMapType() == ignition::common::NormalMapSpace::TANGENT)
+      {
+        pbrWorkflow.SetNormalMap(
+          pbr->NormalMap(), sdf::NormalMapSpace::TANGENT);
+      }
+      else
+      {
+        pbrWorkflow.SetNormalMap(
+          pbr->NormalMap(), sdf::NormalMapSpace::OBJECT);
+      }
+
+      if (pbr->Type() == ignition::common::PbrType::METAL)
+      {
+        pbrOut.SetWorkflow(sdf::PbrWorkflowType::METAL, pbrWorkflow);
+      }
+      else if (pbr->Type() == ignition::common::PbrType::SPECULAR)
+      {
+        pbrOut.SetWorkflow(sdf::PbrWorkflowType::SPECULAR, pbrWorkflow);
+      }
+      out.SetPbrMaterial(pbrOut);
+    }
+    else if (!_in->TextureImage().empty())
+    {
+      sdf::Pbr pbrOut;
+      sdf::PbrWorkflow pbrWorkflow;
+      pbrWorkflow.SetAlbedoMap(_in->TextureImage());
+      pbrOut.SetWorkflow(sdf::PbrWorkflowType::SPECULAR, pbrWorkflow);
+      out.SetPbrMaterial(pbrOut);
+    }
+
+    return out;
+  }
+
+  std::shared_ptr<ignition::common::Material> convert(const sdf::Material &_in)
+  {
+    auto out = std::make_shared<ignition::common::Material>();
+    out->SetEmissive(_in.Emissive());
+    out->SetDiffuse(_in.Diffuse());
+    out->SetSpecular(_in.Specular());
+    out->SetAmbient(_in.Ambient());
+    out->SetRenderOrder(_in.RenderOrder());
+    out->SetLighting(_in.Lighting());
+    out->SetAlphaFromTexture(false, false, _in.DoubleSided());
```

Comment:
According to the API, `SetAlphaFromTexture` takes a `double` for the second input parameter: https://github.com/ignitionrobotics/ign-common/blob/ignition-common4_4.5.0/graphics/include/ignition/common/Material.hh#L161-L166

Here, the second input parameter is a `bool`. The code still compiles because `false` implicitly converts to an int with a value of 0, which is then implicitly converted to a double of `0.0`. I don't think we want to rely on this implicit conversion, though. What if we used the default value for the second parameter (`0.5`) instead? That would make the call look like this:
```suggestion
    out->SetAlphaFromTexture(false, 0.5, _in.DoubleSided());
```